### DISCUSSION TITLE
Welcome to G5API Discussions!

### DISCUSSION THREAD

**Discussion by PhlexPlexico** [PhlexPlexico/G5API#74] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  Also, if you need help setting up the API, please make sure to post in the Help section, and someone may be able to help you out!

**Comment by thboss** [comment]
is there a discord server for chatting?

**Reply by PhlexPlexico** [reply]
Nope, I'd much prefer to keep this project just on GitHub since it's easier to consolidate all problems into one place, instead of having to look in different areas, sorry!

**Comment by stranden** [comment]
Hi @PhlexPlexico,

Do you have any public documentation on how get5-webapi and G5-API are working together?
Just so I will be able understand what G5-API are doing?

Because I want to add an extra layer between a frontend and G5-API.

**Reply by PhlexPlexico** [reply]
Hey! 

Well, I don't have any easy-to-consume documentation... But everything that's under the [legacy](https://phlex.avidgamers.me/api/api-docs/#/legacy) portion of G5API is the calls that are received from the plugin on the CSGO server. Within the get5 plugin, there are a handful of public forwards that can be used, such as `Get5_OnSeriesResult` that would call the [/match/:matchid/finish](https://phlex.avidgamers.me/api/api-docs/#/legacy/post_match__match_id_finish) call. So everything that the API consumes from the game server is just located under there, which requires (confusingly enough) an API key that the system generates per match. 

If you'd like, I could go into more detail once I have a bit more free time explaining what each of the calls do, but from the API Docs and reading the source, I think it should (hopefully!) be self explanatory!

Thanks!

E: And if you'd like to view the latest changes for the WebAPI plugin, they can viewed [here](https://github.com/PhlexPlexico/get5-webapi/blob/utility-knife-dmg/scripting/G5WS.sp)

**Reply by stranden** [reply]
Thanks for the quick introduction! That helped a lot, I will take a look on the legacy part.
You are doing an awesome work!

**Reply by PhlexPlexico** [reply]
Anytime! I'm glad people are making some use of this application :)

**Comment by Bunker56** [comment]
Huge respect! The latest version struck with its speed... I was pleasantly surprised!..
